src/ocm/components/clusterConfiguration/networkConfiguration/NetworkConfigurationForm.tsx

```diff
@@ -0,0 +1,226 @@
+        hostPrefix: defaultNetworkSettings.clusterNetworkHostPrefix,
+        clusterId: cluster.id,
+      },
+    ],
+  };
+
+  const { addAlert, clearAlerts, alerts } = useAlerts();
+  const dispatch = useDispatch();
+  const hostSubnets = React.useMemo(() => getHostSubnets(cluster), [cluster]);
+  const initialValues = React.useMemo(
+    () => getNetworkInitialValues(cluster, defaultNetworkValues),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [], // just once, Formik does not reinitialize
+  );
+
+  const initialTouched = React.useMemo(() => mapValues(initialValues, () => true), [initialValues]);
+
+  const memoizedValidationSchema = React.useMemo(
+    () => getNetworkConfigurationValidationSchema(initialValues, hostSubnets),
+    [hostSubnets, initialValues],
+  );
+
+  React.useEffect(() => {
+    if (infraEnvError) {
+      const title = `Failed to retrieve infra env (clusterId: ${cluster.id})`;
+      //TODO(brotman) add handling of existing errors to alerts context
+      if (alerts.find((alert) => alert.title === title)) {
+        return;
+      }
+      captureException(infraEnvError, title);
+      addAlert({
+        title,
+        message: infraEnvError.message,
+      });
+    }
+    //shouldn't respond to cluster polling. shouldn't respond to alerts changes so remove alert wouldn't trigger adding it back
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [infraEnvError]);
+
+  const handleSubmit: FormikConfig<NetworkConfigurationValues>['onSubmit'] = async (
+    values,
+    actions,
+  ) => {
+    clearAlerts();
+    // update the cluster configuration
+    try {
+      const isMultiNodeCluster = !isSNO(cluster);
+      const isUserManagedNetworking = values.managedNetworkingType === 'userManaged';
+
+      const params: V2ClusterUpdateParams = {
+        apiVip: values.apiVip,
+        ingressVip: values.ingressVip,
+        sshPublicKey: values.sshPublicKey,
+        vipDhcpAllocation: values.vipDhcpAllocation,
+        networkType: values.networkType,
+        machineNetworks: values.machineNetworks,
+        clusterNetworks: values.clusterNetworks,
+        serviceNetworks: values.serviceNetworks,
+        userManagedNetworking: isUserManagedNetworking,
+      };
+
+      if (params.userManagedNetworking) {
+        delete params.apiVip;
+        delete params.ingressVip;
+        if (isMultiNodeCluster) {
+          delete params.machineNetworks;
+        }
+      } else {
+        // cluster-managed can't be chosen in SNO, so this must be a multi-node cluster
+        if (values.vipDhcpAllocation) {
+          delete params.apiVip;
+          delete params.ingressVip;
+        } else if (values.stackType === IPV4_STACK) {
+          delete params.machineNetworks;
+        }
+      }
+
+      const { data } = await ClustersAPI.update(cluster.id, params);
+      actions.resetForm({ values: getNetworkInitialValues(data, defaultNetworkValues) });
```

Comment:
this again introduces the race and hard to follow behavior.
If the latest build passes QE, this should be changed to not reset the form at all & use autosave hook
@jkilzi WDYT ?

Reply:
agree

Reply:
`resetForm` removed.